Classes/New Issue/IssueTemplates.swift

```diff
@@ -0,0 +1,220 @@
+            return []
+        }
+    }
+
+    static func getNameAndDescriptionFromTemplateFile(file: String) -> (name: String?, about: String?) {
+
+        let names = IssueTemplateHelper.matchesForRegexInText(regex: "(?<=name:).*", text: file)
+        let abouts = IssueTemplateHelper.matchesForRegexInText(regex: "(?<=about:).*", text: file)
+        let name = names.count > 0
+            ? names[0].trimmingCharacters(in: .whitespaces)
+            : nil
+        let about = abouts.count > 0
+            ? abouts[0].trimmingCharacters(in: .whitespaces)
+            : nil
+        return (name, about)
+    }
+
+    static func showIssueAlert(
+        with templates: [IssueTemplate],
+        owner: String,
+        repo: String,
+        session: GitHubUserSession?,
+        mainViewController: UIViewController,
+        delegate: NewIssueTableViewControllerDelegate) {
+
+        let alertView = UIAlertController(
+            title: NSLocalizedString("New Issue", comment: ""),
+            message: NSLocalizedString("Choose Template", comment: ""),
+            preferredStyle: .alert
+        )
+
+        for template in templates {
+            alertView.addAction(
+                UIAlertAction(
+                    title: template.title,
+                    style: .default,
+                    handler: { _ in
+
+                        guard let viewController = NewIssueTableViewController.createWithTemplate(
+                            client: GithubClient(userSession: session),
+                            owner: owner,
+                            repo: repo,
+                            template: template.template,
+                            signature: template.title == "Bug Report" ? .bugReport : .sentWithGitHawk
+                            ) else {
+                                Squawk.showGenericError()
+                                return
+                        }
+                        viewController.delegate = delegate
+                        let navController = UINavigationController(rootViewController: viewController)
+                        navController.modalPresentationStyle = .formSheet
+                        mainViewController.present(navController, animated: trueUnlessReduceMotionEnabled)
+                }))
+        }
+
+        alertView.addAction(
+            UIAlertAction(
+                title: "Dismiss",
+                style: UIAlertActionStyle.cancel,
+                handler: { _ in
+                    alertView.dismiss(animated: true, completion: nil)
+            })
+        )
+        mainViewController.present(alertView, animated: true, completion: nil)
+    }
+}
+
+extension GithubClient {
+
+    private func fetchTemplateFile(
+        owner: String,
+        repo: String,
+        filename: String,
+        completion: @escaping (Result<String>) -> Void
+        ) {
+
+        self.fetchFile(
+            owner: owner,
+            repo: repo,
+            branch: "master",
```

Comment:
Is this expected to be hardcoded?

Reply:
I don't think so. I think you can even not have a `master` branch; can we get the default branch somehow? cc @rnystrom

Reply:
@BasThomas "If you create templates on other branches, they will not be available for collaborators to use."
[Link](https://help.github.com/articles/about-issue-and-pull-request-templates/)

Reply:
Actually, if someone knows how to get default branch easily that would be good. I know we have access to it in the branch switcher, not sure if its a quick get or not

Reply:
Figured this out, got it from the RepositoryDetails